spec/java/util/LinkedHashSet.main.lsl

```diff
@@ -241,10 +237,11 @@ automaton LinkedHashSetAutomaton
 
     fun *.iterator (@target self: LinkedHashSet): Iterator
     {
-        val visitedKeysMap: map<Object, Object> = action MAP_NEW();
+        val unseenKeys: map<Object, Object> = action MAP_NEW();
+        action MAP_UNITE_WITH(unseenKeys, this.storage);
```

Comment:
The `MAP_CLONE` action may be helpful here.

Reply:
Fixed;